twill-yarn/src/test/java/org/apache/twill/filesystem/HDFSLocationTest.java

```diff
@@ -55,11 +55,19 @@ protected LocationFactory createLocationFactory(String pathBase) throws Exceptio
 
   // TODO (TWILL-209): figure out how to make MiniDFSCluster enforce permissions
   @Ignore
-  @Test(expected = Exception.class)
   public void testPermissions() throws IOException {
     // create a directory that does not permit anything
     dfsCluster.getFileSystem().mkdir(new Path("/a"), FsPermission.valueOf("----------"));
-    // creating a subdir should fail because even the owner has no write permission
-    dfsCluster.getFileSystem().mkdir(new Path("/a/b"), FsPermission.valueOf("----------"));
+    boolean succeeded = false;
+    try {
+      // creating a subdir should fail because even the owner has no write permission
+      dfsCluster.getFileSystem().mkdir(new Path("/a/b"), FsPermission.valueOf("----------"));
+      succeeded = true;
+    } catch (Exception e) {
+      // expected; TODO: figure out which exception should the expected here, it's not documented
+    }
+    if (succeeded) {
```

Comment:
No it won't I think. Since Assert.fail throws AssertionError which doesn't extend Exception?